contracts/airdropsSMCampaign/Existing1155NftDropSMCampaign.sol

```diff
@@ -0,0 +1,180 @@
+    // to provide this as a bytes32 type and not string. Ox should be prepended.
+    bytes32 public merkleRoot;
+
+    uint256 public immutable airdropDuration;
+    uint256 public immutable airdropStartTime;
+    uint256 public immutable airdropFinishTime;
+
+    modifier onlyAdmin(){
+        if(msg.sender != IAirBroFactory(airBroFactoryAddress).admin()) revert NotAdmin();
+        _;
+    }
+
+    constructor(
+        address _rewardedNft,
+        address _reward1155Nft,
+        uint256 _tokensPerClaim,
+        uint256 _tokenId,
+        uint256 _totalAirdropAmount,
+        uint256 _airdropDuration,
+        address _airBroFactoryAddress
+    ) {
+        rewardedNft = IERC1155(_rewardedNft);
+        tokensPerClaim = _tokensPerClaim;
+        totalAirdropAmount = _totalAirdropAmount;
+        rewardToken = IERC1155(_reward1155Nft);
+        rewardTokenId = _tokenId;
+        airdropDuration = _airdropDuration * 1 days;
+        airdropStartTime = block.timestamp;
+        airdropFinishTime = block.timestamp + airdropDuration;
+        airBroFactoryAddress = _airBroFactoryAddress;
+
+    }
+
+    /// @notice Sets the merkleRoot - can only be done if admin (different from the contract owner)
+    /// @param _merkleRoot - The root hash of the Merle Tree
+    function setMerkleRoot(bytes32 _merkleRoot) external onlyAdmin {
+        merkleRoot = _merkleRoot;
+        emit MerkleRootChanged(_merkleRoot);
+    }
+
+    /// @notice Allows the airdrop creator to provide funds for airdrop reward
+    function fundAirdrop() external {
+        if (rewardToken.balanceOf(msg.sender, rewardTokenId) < totalAirdropAmount) revert InsufficientAmount();
+        if (airdropFunded) revert AlreadyFunded();
+        rewardToken.safeTransferFrom(msg.sender, address(this), rewardTokenId, totalAirdropAmount, "");
+        airdropFunded = true;
+        airdropFundBlockTimestamp = block.timestamp;
+        airdropFundingHolder = msg.sender;
+        emit AirdropFunded(address(this));
+    }
+
+    /// @notice Allows the airdrop creator to withdraw back his funds after the airdrop has finished
+    function withdrawAirdropFunds() external {
+        if (airdropFundingHolder != msg.sender) revert NotOwner();
+        if (block.timestamp < airdropFinishTime) revert AirdropStillInProgress();
+        rewardToken.safeTransferFrom(
+            address(this),
+            msg.sender,
+            rewardTokenId,
+            rewardToken.balanceOf(address(this), rewardTokenId),
+            ""
+        );
+    }
+
+    /// @notice Allows the NFT holder to claim his ERC1155 airdrop
+    function claim(bytes32[] calldata _merkleProof) external {
+        if (hasClaimed[msg.sender]) revert AlreadyRedeemed();
+        if (rewardToken.balanceOf(address(this), rewardTokenId) < tokensPerClaim) revert InsufficientLiquidity();
+        if (block.timestamp > airdropFinishTime) revert AirdropExpired();
+
+        bool isEligible = checkProof(_merkleProof, merkleRoot);
```

Comment:
You can call `isEligibleForReward` function here. The code is duplicated
`bool isEligible = isEligibleForReward(_merkleProof)`
...